### ISSUE TITLE
no type named 'backendCos1'

### ISSUE THREAD

**Issue by mohsensadr** [IPPL-framework/ippl#255]
Compilation error on Mac and Merlin

` ./999-build-everything -t serial --kokkos --heffte --ippl -u`

```
ippl/src/FFT/FFT.h:291:11: error: no type named 'backendCos1' in 'struct ippl::detail::HeffteBackendType<Kokkos::HostSpace>'
  291 |     class FFT<Cos1Transform, Field> : public IN_PLACE_FFT_BASE_CLASS(Field, backendCos1) {
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~
/psi/home/sadr_m/Codes/ippl_master/ippl/src/FFT/FFT.h:293:15: error: no type named 'backendCos1' in 'struct ippl::detail::HeffteBackendType<Kokkos::HostSpace>'
  293 |         using Base                    = IN_PLACE_FFT_BASE_CLASS(Field, backendCos1);
      |               ^~~~
...
```

https://github.com/IPPL-framework/ippl/blob/8967b6e10c56db6182ac1d364d47bb45bc96d476/src/FFT/FFT.h#L291

**Comment by Arc676** [comment]
For reference, this compilation error has nothing to do with the platform. It is solely affected by the choice of FFT backends and whether the backend for DCT1 is present under the chosen configuration, which it is not under the `serial` configuration.

**Comment by s-mayani** [comment]
I will close this issue since the fix is already in dev-3.2.0.
